sdk/keyvault/azkeys/testdata/perf/decrypt.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/perf"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azkeys/crypto"
+)
+
+type decryptTestOptions struct{}
+
+var decryptTestOpts decryptTestOptions = decryptTestOptions{}
+
+type decryptTest struct {
+	perf.PerfTestOptions
+	keyName      string
+	client       *azkeys.Client
+	cryptoClient *crypto.Client
+	encrypAlg    crypto.EncryptionAlg
+	cipherText   []byte
+}
+
+// newDecryptTest is called once per process
+func newDecryptTest(ctx context.Context, options perf.PerfTestOptions) (perf.GlobalPerfTest, error) {
+	d := &decryptTest{
+		PerfTestOptions: options,
+		keyName:         "livekvtestdecryptperfkey",
+		encrypAlg:       crypto.EncryptionAlgRSAOAEP256,
+	}
+
+	vaultURL, ok := os.LookupEnv("AZURE_KEYVAULT_URL")
+	if !ok {
+		return nil, fmt.Errorf("the environment variable 'AZURE_KEYVAULT_URL' could not be found")
+	}
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	client, err := azkeys.NewClient(vaultURL, cred, &azkeys.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: options.Transporter,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.CreateRSAKey(ctx, d.keyName, nil)
```

Comment:
To align with .NET, at least, I'd specify a key size of 2048, unless that's already your default. I presume, though, you let the service default which can change and change the perf measurements. I'd hardcode it to 2048.